expressjs-backend/util/s3.test.js

```diff
@@ -0,0 +1,82 @@
+const s3 = require("./s3");
+
+beforeEach(async () => {
+    //Ensure clean testing folder before each test.
+    await s3.deleteS3Folder("testing");
+});
+
+afterAll(async () => {
+    //Cleanup after test suite runs.
+    await s3.deleteS3Folder("testing");
+});
+
+test("Test successful file upload", async () => {
+    const imageName = "testing/testImage";
+    const imageData = "testimagedata";
+    //Returns the AWS response which is the link to the uploaded file if success
+    const response = await s3.uploadImageToS3(imageName, imageData);
+    
+    //Ensure link is reflective of uploaded file
+    expect(response).toEqual(`https://haggleimgs.s3.amazonaws.com/${imageName}`);
+});
+
+
+test("Test unsuccessful file upload(null image data", async () => {
+    const imageName = "testing/testImage";
+    const imageData = null;
+    await expect(s3.uploadImageToS3(imageName, imageData)).rejects.toThrow();
+});
+
+test("Test unsuccessful file upload(null image data", async () => {
+    const imageName = null;
+    const imageData = "imagedata";
+    await expect(s3.uploadImageToS3(imageName, imageData)).rejects.toThrow();
+});
+
+test("Test Successful listing of objects", async () => {
+    //Upload testing file
+    const imageName = "testing/testImage";
+    const imageData = "testimagedata";
+    await s3.uploadImageToS3(imageName, imageData);
+
+    const testFiles = await s3.listS3Objects("testing");
+    expect(testFiles.length > 0).toEqual(true);
```

Comment:
Great test, but maybe you should test with an expected length value rather than > 0 to ensure the objects were not listed too many times (like a double upload).

Reply:
Good idea, will change